### ISSUE TITLE
Improve prerequisites for “Run a Replicated Stateful Application”

### ISSUE THREAD

**Issue by sdwerwed** [kubernetes/website#24750]
Replication doesnt work by following this https://kubernetes.io/docs/tasks/run-application/run-replicated-stateful-application/
 slave can not connect to master. (There is no replication user in mysql.user) I used namespace: mysql for all yml
```
mysql> SHOW SLAVE STATUS\G;
*************************** 1. row ***************************
               Slave_IO_State: Connecting to master
                  Master_Host: mysql-0.mysql
                  Master_User: root
                  Master_Port: 3306
                Connect_Retry: 10
              Master_Log_File: mysql-0-bin.000010
          Read_Master_Log_Pos: 154
               Relay_Log_File: mysql-2-relay-bin.000004
                Relay_Log_Pos: 4
        Relay_Master_Log_File: mysql-0-bin.000010
             Slave_IO_Running: Connecting
            Slave_SQL_Running: Yes
              Replicate_Do_DB: 
          Replicate_Ignore_DB: 
           Replicate_Do_Table: 
       Replicate_Ignore_Table: 
      Replicate_Wild_Do_Table: 
  Replicate_Wild_Ignore_Table: 
                   Last_Errno: 0
                   Last_Error: 
                 Skip_Counter: 0
          Exec_Master_Log_Pos: 154
              Relay_Log_Space: 154
              Until_Condition: None
               Until_Log_File: 
                Until_Log_Pos: 0
           Master_SSL_Allowed: No
           Master_SSL_CA_File: 
           Master_SSL_CA_Path: 
              Master_SSL_Cert: 
            Master_SSL_Cipher: 
               Master_SSL_Key: 
        Seconds_Behind_Master: NULL
Master_SSL_Verify_Server_Cert: No
                Last_IO_Errno: 1045
                Last_IO_Error: error connecting to master 'root@mysql-0.mysql:3306' - retry-time: 10  retries: 300
               Last_SQL_Errno: 0
               Last_SQL_Error: 
  Replicate_Ignore_Server_Ids: 
             Master_Server_Id: 0
                  Master_UUID: 
             Master_Info_File: /var/lib/mysql/master.info
                    SQL_Delay: 0
          SQL_Remaining_Delay: NULL
      Slave_SQL_Running_State: Slave has read all relay log; waiting for more updates
           Master_Retry_Count: 86400
                  Master_Bind: 
      Last_IO_Error_Timestamp: 201027 10:20:56
     Last_SQL_Error_Timestamp: 
               Master_SSL_Crl: 
           Master_SSL_Crlpath: 
           Retrieved_Gtid_Set: 
            Executed_Gtid_Set: 
                Auto_Position: 0
         Replicate_Rewrite_DB: 
                 Channel_Name: 
           Master_TLS_Version: 
1 row in set (0.00 sec)
```

**Comment by sftim** [comment]
/retitle Replication not working in “Run a Replicated Stateful Application”

**Comment by sftim** [comment]
Thanks for reporting this @sdwerwed - we'll check it out.

**Comment by sdwerwed** [comment]
Changing the namespece form `mysql` to somethink else like `mysqlns` works

**Comment by sftim** [comment]
Hi @sdwerwed I think what you're saying is that in your cluster this didn't work if the Kubernetes namespace you deploy it into is named `mysql`, but does work if you use a different (non-reserved) namespace name.

Have I got that right?

**Comment by sdwerwed** [comment]
> Hi @sdwerwed I think what you're saying is that in your cluster this didn't work if the Kubernetes namespace you deploy it into is named `mysql`, but does work if you use a different (non-reserved) namespace name.
> 
> Have I got that right?

Hello, yes you got it 👍

**Comment by sftim** [comment]
Let's document that this example assumes you're using the _default_ namespace or another namespace that doesn't contain any conflicting objects.

**Comment by sftim** [comment]
/retitle Improve prerequisites for “Run a Replicated Stateful Application”
/triage accepted
/language en

**Comment by sdwerwed** [comment]
Closed as #27128 is merged
